packages/server/src/api/controllers/row/ExternalRequest.ts

```diff
@@ -248,6 +163,117 @@ export class ExternalRequest<T extends Operation> {
     if (datasource && datasource.entities) {
       this.tables = datasource.entities
     }
+    this.tableList = Object.values(this.tables)
+  }
+
+  private prepareFilters(
+    id: string | undefined | string[],
+    filters: SearchFilters,
+    table: Table
+  ) {
+    const tables = this.tableList
+    // replace any relationship columns initially, table names and relationship column names are acceptable
+    const relationshipColumns = sdk.rows.filters.getRelationshipColumns(table)
+    filters = sdk.rows.filters.updateFilterKeys(
+      filters,
+      relationshipColumns
+        .map(({ name, definition }) => {
+          const { tableName } = breakExternalTableId(definition.tableId)
+          return {
+            original: name,
+            updated: tableName!,
+          }
+        })
+        // don't update table names - include this for context incase a column would be replaced
+        .concat(
+          tables.map(table => {
+            const tableName = table.originalName || table.name
+            return {
+              original: tableName,
+              updated: tableName,
+            }
+          })
+        )
```

Comment:
I'm not sure that I understand the purpose of this. Was this in place for when we were doing the sorting? And now that we've fixed that, it's not necessary?